### PR TITLE
Add events to the performance profile controller

### LINKED ISSUE #29: Add events to the performance profile controller

### PR DESCRIPTION
**Pull request by cynepco3hahue**
Fixes #29 

Signed-off-by: Artyom Lukianov <alukiano@redhat.com>

### CODE REVIEW

File: pkg/controller/performanceprofile/performanceprofile_controller.go [hunk 1]
```diff
@@ -304,6 +313,7 @@ func (r *ReconcilePerformanceProfile) Reconcile(request reconcile.Request) (reco
 		return *result, nil
 	}
 
+	r.recorder.Eventf(instance, corev1.EventTypeNormal, "Creation succeeded", "Succeeded to create all components")
```
Comment (davidvossel) [inline pkg/controller/performanceprofile/performanceprofile_controller.go]: won't this event get posted every time the reconcile loop pops after everything is stable?

it seems like a better option might be to post this in the applyComponents function only in the situation that the final component change was made. so not when all components already exist, but when we actually finish successfully posting all components. 

that way the event only triggers once, or only one per a mutation of the profile object.

File: pkg/controller/performanceprofile/performanceprofile_controller_test.go [hunk 2]
```diff
@@ -249,6 +256,12 @@ var _ = Describe("Controller", func() {
 			result, err = r.Reconcile(request)
 			Expect(err).ToNot(HaveOccurred())
 			Expect(result).To(Equal(reconcile.Result{}))
+
+			// verify creation event
+			fakeRecorder, ok := r.recorder.(*record.FakeRecorder)
+			Expect(ok).To(BeTrue())
+			event := <-fakeRecorder.Events
+			Expect(event).To(ContainSubstring("Creation succeeded"))
```
Comment (davidvossel) [inline pkg/controller/performanceprofile/performanceprofile_controller_test.go]: can we get a func test that ensures that if all objects already exist, and the reconcile loop is poped, that no events are triggered?

Comment (cynepco3hahue) [reply pkg/controller/performanceprofile/performanceprofile_controller_test.go]: func or unit test?

Comment (cynepco3hahue) [reply pkg/controller/performanceprofile/performanceprofile_controller_test.go]: done

File: pkg/controller/performanceprofile/performanceprofile_controller.go (hunk 1 above)
Comment (cynepco3hahue) [reply pkg/controller/performanceprofile/performanceprofile_controller.go]: done

File: pkg/controller/performanceprofile/performanceprofile_controller_test.go (hunk 2 above)
Comment (davidvossel) [reply pkg/controller/performanceprofile/performanceprofile_controller_test.go]: oh, i meant unit. sorry. you figured it out though :+1:

**Comment by davidvossel** [comment]
pr looks ready to go. I just want to re-kick the e2e tests and see them pass first. 

interesting e2e test error. I don't think it's relatd to this PR, but maybe there's something we should be doing to be more resilient to failures to reach the api server. 

```
Activated service account credentials for: [ci-provisioner@openshift-gce-devel-ci.iam.gserviceaccount.com]
Updated property [core/project].
/go/src/github.com/openshift-kni/performance-addon-operators
Downloading kustomize v3.5.3
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   626    0   626    0     0   3130      0 --:--:-- --:--:-- --:--:--  3130

100 10.7M  100 10.7M    0     0  13.2M      0 --:--:-- --:--:-- --:--:-- 13.2M
kustomize
Deploying operator
FULL_REGISTRY_IMAGE=""quay.io"/""/"performance-addon-operators-registry":"latest"" KUSTOMIZE="""_cache"/tools"/"kustomize"" hack/deploy.sh
Openshift CI detected, deploying using image registry.svc.ci.openshift.org/ci-op-06gyxm6f/stable:performance-addon-operators-registry
error: unable to recognize "STDIN": Get https://api.ci-op-06gyxm6f-24cc7.origin-ci-int-gce.dev.openshift.com:6443/api?timeout=32s: dial tcp: lookup api.ci-op-06gyxm6f-24cc7.origin-ci-int-gce.dev.openshift.com on 10.142.0.111:53: no such host
make: *** [Makefile:115: cluster-deploy] Error 1
```

**Comment by slintes** [comment]
> interesting e2e test error. I don't think it's relatd to this PR, but maybe there's something we should be doing to be more resilient to failures to reach the api server.
> 
I'm already changing deployment to the "repeat-until-success" way with kustomize in the MCP PR

**Pull request merged**